migrations/Version202108240432573635_taoPublishing.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace oat\taoPublishing\migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use oat\tao\scripts\tools\migrations\AbstractMigration;
+use oat\tao\model\search\SearchProxy;
+use oat\taoPublishing\model\PlatformService;
+
+final class Version202108240432573635_taoPublishing extends AbstractMigration
+{
+
+    public function getDescription(): string
+    {
+        return 'Register Remote Environment class URI to use generic search in ' . SearchProxy::class;
+    }
+
+    public function up(Schema $schema): void
+    {
+        $generisSearchWhitelist = [
+            PlatformService::CLASS_URI,
+        ];
+        $searchProxy = $this->getProxy();
+        
+        if ($searchProxy->hasOption(SearchProxy::OPTION_GENERIS_SEARCH_WHITELIST)) {
+            $options = $searchProxy->getOption(SearchProxy::OPTION_GENERIS_SEARCH_WHITELIST);
+            $generisSearchWhitelist = array_merge($options, $generisSearchWhitelist);
+        }
```

Comment:
Why do we need an if `hasOption`, would it be possible to happen if we will force a new tao-core version?

Reply:
How you can make it more readable and simpler: https://github.com/oat-sa/extension-tao-publishing/pull/115/files#r695439951